pkg/controllers/scalablenodegroup/v1alpha1/suite_test.go

```diff
@@ -53,24 +54,87 @@ var _ = AfterSuite(func() {
 
 var _ = Describe("Examples", func() {
 	var ns *environment.Namespace
+	var desiredReplicas = ptr.Int32(5)
+	var dummyMessage = "test message"
+	var nodeGroupObject *fake.NodeGroup
 	var sng *v1alpha1.ScalableNodeGroup
+	var ng cloudprovider.NodeGroup
+	var defaultNodeGroupObject fake.NodeGroup
 
 	BeforeEach(func() {
 		var err error
+		sng = &v1alpha1.ScalableNodeGroup{}
+		ng = fakeController.CloudProvider.NodeGroupFor(&sng.Spec)
+		defaultNodeGroupObject = *ng.(*fake.NodeGroup)
 		ns, err = env.NewNamespace()
 		Expect(err).NotTo(HaveOccurred())
-		sng = &v1alpha1.ScalableNodeGroup{}
+		sng.Spec.Replicas = desiredReplicas
+		nodeGroupObject = ng.(*fake.NodeGroup)
+	})
+
+	AfterEach(func() {
+		nodeGroupObject.Stable = defaultNodeGroupObject.Stable
+		nodeGroupObject.Message = defaultNodeGroupObject.Message
+		nodeGroupObject.WantErr = defaultNodeGroupObject.WantErr
+		sng.Spec.Replicas = desiredReplicas
+		ng = fakeController.CloudProvider.NodeGroupFor(&sng.Spec)
 	})
 
 	Context("ScalableNodeGroup", func() {
 		It("should be created", func() {
 			Expect(ns.ParseResources("docs/examples/reserved-capacity-utilization.yaml", sng)).To(Succeed())
-			sng.Spec.Replicas = ptr.Int32(5)
-
 			ExpectCreated(ns.Client, sng)
 			ExpectEventuallyHappy(ns.Client, sng)
-
 			ExpectDeleted(ns.Client, sng)
 		})
 	})
+
+	Context("Basic ScalableNodeGroup Reconcile tests", func() {
+		It("Test reconciler to scale up nodes", func() {
+			Expect(fakeController.Reconcile(sng)).To(Succeed())
+			Expect(ng.GetReplicas()).To(Equal(*desiredReplicas))
+		})
+
+		It("Test reconciler to scale down nodes", func() {
+			Expect(ng.SetReplicas(10)).To(Succeed()) // set existing replicas higher than desired
+			Expect(fakeController.Reconcile(sng)).To(Succeed())
+			Expect(ng.GetReplicas()).To(Equal(*desiredReplicas))
+		})
+
+		It("Test reconciler to make no change to node count", func() {
+			Expect(ng.SetReplicas(*desiredReplicas)).To(Succeed()) // set existing replicas equal to desired
+			Expect(fakeController.Reconcile(sng)).To(Succeed())
+			Expect(ng.GetReplicas()).To(Equal(*desiredReplicas))
+		})
+	})
+
+	Context("Advanced ScalableNodeGroup Reconcile tests", func() {
```

Comment:
I'm not sure I see the difference between Basic and Advanced tests, they share all Before/After hooks and neither context has variables. Does it make sense to just collapse them?

Reply:
Yes, when I started I though I will be adding some complex test but it didn't make sense while writing tests. So yes, I will combine them.